# Ejercicio 13

## Rosas Hernández Ariel Jesús

Lavado Automata para automóviles funciona sólo con un lugar. Los autos llegan siguiendo una distribución de Poisson, con 4 autos por hora, que pueden esperar en el estacionamiento de la instalación, si el lugar de lavado está ocupado. El tiempo para lavar y limpiar un automóvil es exponencial, con 10 minutos de promedio. Los automóviles que no se pueden estacionar en la instalación pueden esperar en el arrollo junto al lavado. Eso quiere decir que para todo fin práctico no hay límite del tamaño del sistema. El gerente de la instalación desea determinar el tamaño del estacionamiento.

Para resolver este problema, hay que determinar $L_q$, pero de momento estimemos $L$ y $W$ usando la simulación.

Calcular $L_q$ para el ejemplo anterior, usando las fórmulas de arriba

$\lambda=4$
$\mu=6$

In [2]:
from random import expovariate
from statistics import mean 
from math import inf as Infinity
# Parametros

lamda = 4.0                         # Razón de arrivos (1/Lambda)  Ojo
mu = 6.0                            # Razón de salida (1/Mu)       Ojo


Num_Pkts = 100000                   # Numero de paquetes a simular
#T = 100000
count = 0                           # Contador de paquetes simulados
t = 0
N = 0                               # Variable de estado: numero de paquetes en el sistema

tll = expovariate(lamda)            # Tiempo de llegada
ts = Infinity                       # Tiempo de salida

t_evento_anterior = 0.0             # tiempo del último evento ****************Nuevas líneas de código***********


# Variables de salida
tll_Data = []                       # Lista de tiempos de llegada
ts_Data = []                        # Lista de tiempos de salida
r_Data = []                         # Lista de retrasos individuales
Area = []                           # Area de escalón ***********************************************************


while count < Num_Pkts:
#while t < T:
    if tll < ts:                           # Evento de llegada
        t = tll
        tll_Data.append(t)
        
        # Area del escalón ***********************************************************************************
        Area.append((t - t_evento_anterior)*N)
        t_evento_anterior = t
        
        
        N = N + 1.0
        tll = t + expovariate(lamda) 
        if N == 1:
            ts = t + expovariate(mu)
    else:                                         # Evento de salida
        t = ts
        ts_Data.append(t)
        
        # Area del escalón ***********************************************************************************
        Area.append((t - t_evento_anterior)*N)
        t_evento_anterior = t
        
        
        N = N - 1.0
        count = count + 1                         # Paquetes simulados
        if N > 0:
            ts = t + expovariate(mu)
        else:
            ts = Infinity
            
# Estimación del retraso promedio:

for i in range(Num_Pkts):
#for i in range(T):
    d = ts_Data[i] - tll_Data[i]
    r_Data.append(d)

print( "Retraso promedio = W = ", round( mean(r_Data), 4) )
print( "E[ N(t)] = L = ", round(sum(Area)/t, 4))

Retraso promedio = W =  0.4915
E[ N(t)] = L =  1.9657


Sabemos por la simulacion que...

$L=1.9657$

$W=0.4915$

$L_q=\lambda W_q$

Donde

$W_q=W-\frac{1}{\mu}$

Por lo tanto 

$W_q=0.4915-\frac{1}{6}$=0.324833$

$L_q= \lambda W_q= 4*0.324833=1.29933$